pypeit/find_objects.py

```diff
@@ -1120,6 +1119,68 @@ def illum_profile_spectral(self, global_sky, skymask=None):
         # Now apply the correction to the science frame
         self.apply_relative_scale(scaleImg)
 
+    def convolve_skymodel(self, input_img, fwhm_map, thismask, subpixel=5, nsample=10):
```

Comment:
This routine should probably be moved to a core method wrapped by the method in the class. You are performing core functionality here as far as I can tell.